Running pytest during Travis build fails with ModuleNotFoundError: No module named 'google'

# Summary

Running `poetry run pytest` fails during build because GCP dependencies are not installed.

## How to reproduce

1. Clone repo
1. Install dependencies using `poetry install -E yaml -E toml -E azure -E aws` (just like in `.travis.yml` file)
1. Run `poetry run pytest` 

```
====================================== test session starts =======================================
platform darwin -- Python 3.8.5, pytest-5.4.3, py-1.9.0, pluggy-0.13.1
rootdir: /Users/vduseev/Projects/python-configuration, inifile: setup.cfg
plugins: flake8-1.0.6, mock-2.0.0, mypy-0.4.2, cov-2.10.1, black-0.3.11
collected 200 items / 1 error / 199 selected                                                     

Running mypy on 30 files... done with status 1
Found 56 errors in 1 file (checked 30 source files)

============================================= ERRORS =============================================
___________________________ ERROR collecting tests/contrib/test_gcp.py ___________________________
ImportError while importing test module '/Users/vduseev/Projects/python-configuration/tests/contrib/test_gcp.py'.
Hint: make sure your test modules/packages have valid Python names.
Traceback:
tests/contrib/test_gcp.py:4: in <module>
    from google.api_core.exceptions import NotFound
E   ModuleNotFoundError: No module named 'google'

==================================== short test summary info =====================================
ERROR tests/contrib/test_gcp.py
!!!!!!!!!!!!!!!!!!!!!!!!!!!!! Interrupted: 1 error during collection !!!!!!!!!!!!!!!!!!!!!!!!!!!!!
======================================== 1 error in 2.67s ========================================
```

## What happens

1. `google` module cannot be imported because it is not installed.

## What should happen

1. Successful import.

## Fix suggestions

1. Add `-E gcp` to `.travis.yml` during dependency installation using poetry

--- a/.travis.yml
+++ b/.travis.yml
@@ -11,7 +11,7 @@ before_install:
 - source $HOME/.poetry/env
 
 install:
-- poetry install -E yaml -E toml -E azure -E aws
+- poetry install -E yaml -E toml -E azure -E aws -E gcp
 
 script:
 - poetry run pytest

--- a/.travis.yml
+++ b/.travis.yml
@@ -6,7 +6,9 @@ python:
 - 3.8
 
 before_install:
-- pip install poetry
+- curl -sSL https://raw.githubusercontent.com/python-poetry/poetry/master/get-poetry.py > /tmp/get-poetry.py
+- python /tmp/get-poetry.py --preview --yes
+- source $HOME/.poetry/env
 
 install:
 - poetry install -E yaml -E toml -E azure -E aws

--- a/config/configuration.py
+++ b/config/configuration.py
@@ -3,6 +3,7 @@
 import base64
 from contextlib import contextmanager
 from copy import deepcopy
+from sys import version_info
 from typing import (
     Any,
     Dict,
@@ -26,6 +27,9 @@ from .helpers import (
     clean,
     interpolate_object,
 )
+
+if version_info < (3, 8):
+    from collections import OrderedDict
 
 
 class Configuration:
@@ -299,7 +303,10 @@ class Configuration:
         return iter(dict(self.items()))  # type: ignore
 
     def __reversed__(self) -> Iterator[Tuple[str, Any]]:  # noqa: D105
-        return reversed(dict(self.items()))  # type: ignore
+        if version_info < (3, 8):
+            return OrderedDict(reversed(list(self.items())))  # type: ignore
+        else:
+            return reversed(dict(self.items()))  # type: ignore
 
     def __len__(self) -> int:  # noqa: D105
         return len(self.keys())

--- a/tests/test_interpolation.py
+++ b/tests/test_interpolation.py
@@ -46,7 +46,10 @@ def test_interpolation():  # type: ignore
     assert cfg.var1 == "This is a test"
 
     cfg = config_from_dict(
-        VALUES, lowercase_keys=True, interpolate=True, interpolate_type="unknown",
+        VALUES,
+        lowercase_keys=True,
+        interpolate=True,
+        interpolate_type="unknown",
     )
 
     with raises(ValueError, match='Invalid interpolation method "unknown"'):

